src/rpc/util.cpp

```diff
@@ -846,6 +846,10 @@ void RPCResult::ToSections(Sections& sections, const OuterType outer_type, const
         sections.PushSection({indent + "}" + maybe_separator, ""});
         return;
     }
+    case Type::EMPTY: {
+        sections.PushSection({indent + maybe_key + "{}", Description("empty JSON object")});
```

Comment:
why does this need a new case? Why can't the above case be re-used?

Reply:
Because of `CHECK_NONFATAL(!m_inner.empty());`

The rest of the `Type::OBJ` code block doesn't do anything useful for an empty object either, so it seemed more readable to just add a new case. But I could use an `if` statement instead.

Reply:
I found a more compact solution...

Reply:
Reverting back to using a separate case. The string manipulation code Type:OBJ does not behave nicely if the object is empty, see https://github.com/bitcoin/bitcoin/pull/23706#pullrequestreview-833917359